osfoffline/gui/qt/preferences.py

```diff
@@ -84,12 +89,22 @@ def set_containing_folder(self):
             return
 
         user = Session().query(User).one()
+
+        if save_setting:
+            # copy the synced folders from the old to new location
+            # so OSF doesn't think they were deleted
+            copy_tree(user.folder, os.path.join(osf_path))
```

Comment:
If I put some projects which is not in the sync list or just some random folders inside the original path '/User/Desktop/OSF' for example. It will break as it can't recognize these unrelated folder which it should not copy and also if I choose the new destination to be '/User/Desktop/OSF/OSF' it will copy a new OSF folder into the current directory.